kore/src/main/scala/org/kframework/definition/outer.scala

```diff
@@ -409,18 +409,56 @@ trait Sentence extends HasLocation with HasAtt with AttValue {
   def label: Optional[String] = att.getOptional(Att.LABEL)
 }
 
+object Sentence {
+  implicit val ord = new Ordering[Sentence] {
+    def compare(a: Sentence, b: Sentence): Int = {
+      (a, b) match {
+        case (_:SyntaxSort, _) => -1
```

Comment:
This doesn't feel quite right. Shouldn't it have a case where both are the same and it compares their contents?

Reply:
I had them there but removing them didn't break the unit tests I wrote, so I figured when they're the same type then the ordering for that specific type is used instead of going up the type heirarchy.

Reply:
Looks like we do need those extra cases. I pushed the fix.